DataFormats/SiStripDetId/interface/SiStripDetId.h

```diff
@@ -112,65 +112,80 @@ class SiStripDetId : public DetId {
 // ---------- inline methods ----------
 
 SiStripDetId::SubDetector SiStripDetId::subDetector() const {
-  if( det() == DetId::Tracker &&
-      subdetId() == static_cast<int>(SiStripDetId::TIB) ) {
-    return SiStripDetId::TIB;
-  } else if ( det() == DetId::Tracker &&
-	      subdetId() == static_cast<int>(SiStripDetId::TID) ) {
-    return SiStripDetId::TID;
-  } else if ( det() == DetId::Tracker &&
-	      subdetId() == static_cast<int>(SiStripDetId::TOB) ) {
-    return SiStripDetId::TOB;
-  } else if ( det() == DetId::Tracker &&
-	      subdetId() == static_cast<int>(SiStripDetId::TEC) ) {
-    return SiStripDetId::TEC;
+  SiStripDetId::SubDetector result;
+  if ( det() == DetId::Tracker) {
+    if ( subdetId() == static_cast<int>(SiStripDetId::TEC) ) {
+      result = SiStripDetId::TEC;
+    } else if ( subdetId() == static_cast<int>(SiStripDetId::TID) ) {
+      result = SiStripDetId::TID;
+    } else if ( subdetId() == static_cast<int>(SiStripDetId::TOB) ) {
+      result = SiStripDetId::TOB;
+    } else if ( subdetId() == static_cast<int>(SiStripDetId::TIB) ) {
+      result = SiStripDetId::TIB;
+    } else {
+      result = SiStripDetId::UNKNOWN;
+    }
   } else {
-    return SiStripDetId::UNKNOWN;
+    result = SiStripDetId::UNKNOWN;
   }
+  return result;
 }
 
 SiStripDetId::ModuleGeometry SiStripDetId::moduleGeometry() const {
+  SiStripDetId::ModuleGeometry geometry;
   switch(subDetector()) {
-  case TIB: return int((id_>>layerStartBit_) & layerMask_)<3? IB1 : IB2;
-  case TOB: return int((id_>>layerStartBit_) & layerMask_)<5? OB2 : OB1;
+  case TIB: geometry = int((id_>>layerStartBit_) & layerMask_)<3? IB1 : IB2;
+    break;
+  case TOB: geometry = int((id_>>layerStartBit_) & layerMask_)<5? OB2 : OB1;
+    break;
   case TID: switch ((id_>>ringStartBitTID_) & ringMaskTID_) {
-    case 1: return W1A;
-    case 2: return W2A;
-    case 3: return W3A;
+    case 1: geometry = W1A;
+      break;
+    case 2: geometry = W2A;
+      break;
+    case 3: geometry = W3A;
+      break;
     }
+    break;
   case TEC: switch ((id_>>ringStartBitTEC_) & ringMaskTEC_) {
-    case 1: return W1B;
-    case 2: return W2B;
-    case 3: return W3B;
-    case 4: return W4;
-    case 5: return W5;
-    case 6: return W6;
-    case 7: return W7;
+    case 1: geometry = W1B;
+      break;
+    case 2: geometry = W2B;
+      break;
+    case 3: geometry = W3B;
+      break;
+    case 4: geometry = W4;
+      break;
+    case 5: geometry = W5;
+      break;
+    case 6: geometry = W6;
+      break;
+    case 7: geometry = W7;
+      break;
     }
-  case UNKNOWN: default: return UNKNOWNGEOMETRY;
+    break;
+  case UNKNOWN: default: geometry = UNKNOWNGEOMETRY;
   }
+  return geometry;
 }
```

Comment:
The compiler is complaining that if ```subDetector()``` is either ```TID``` or ```TEC``` and the next level ```switch``` statement doesn't match any of the next level ```case``` statements then ```geometry``` is not set. The easiest solution would be to just set ```SiStripDetId::ModuleGeometry geometry=UNKNOWNGEOMETRY;``` at the very beginning.
```